src/debug/debug.cc

```diff
@@ -3617,6 +3617,13 @@ bool RecordReplayHasRegisteredScript(Script script) {
     gRegisteredScripts->find(script.id()) != gRegisteredScripts->end();
 }
 
+bool RecordReplayIsDivergentUserJSWithoutPause(
+    const SharedFunctionInfo& shared) {
+  return recordreplay::AreEventsDisallowed() &&
+         !recordreplay::HasDivergedFromRecording() &&
+         shared.IsUserJavaScript() && shared.HasSourceCode();
```

Comment:
This is a preexisting issue but I think checking to see if the script has been registered with the recorder would be better and more coherent.  Scripts which haven't been registered are not user JS, but it's not obvious that IsUserJavaScript() will return false for replay specific scripts that don't advance the progress counter (and which we don't consider to be user JS).

Reply:
You recently added the `RecordReplayHasRegisteredScript` check to the `Invoke` code.
I now moved it into `RecordReplayIsDivergentUserJSWithoutPause` and removed other useless checks.